.github/workflows/pythonapp.yml

```diff
@@ -72,3 +72,34 @@ jobs:
           git add --all
           git commit --allow-empty -m "C++ FEniCS/basix@${{ github.sha }}"
           git push
+
+  build-not-dev:
```

Comment:
What flags are different for this build case? Ideally we'd pick up these issues on a single build and run.

Reply:
I doesn't have `-DCMAKE_BUILD_TYPE=Developer`. For reasons beyond me some errors were only happening when building without developer mode (see eg https://github.com/bempp/bempp-cl/runs/2110771847)

Reply:
The CMake build type sets the flags - need to understand which compiler flags need to be set.